ipld/kamt/tests/kamt_tests.rs

```diff
@@ -50,29 +50,74 @@ impl KamtFactory {
 fn test_basics(factory: KamtFactory) {
     let store = MemoryBlockstore::default();
     let mut kamt: HKamt<_, _> = factory.new(&store);
-    kamt.set(1, "world".to_string()).unwrap();
 
+    kamt.set(1, "world".to_string()).unwrap();
     assert_eq!(kamt.get(&1).unwrap(), Some(&"world".to_string()));
+    assert!(kamt.contains_key(&1).unwrap());
+
     kamt.set(1, "world2".to_string()).unwrap();
     assert_eq!(kamt.get(&1).unwrap(), Some(&"world2".to_string()));
+    assert!(kamt.contains_key(&1).unwrap());
+
+    assert_eq!(kamt.get(&2).unwrap(), None);
+    assert!(!kamt.contains_key(&2).unwrap());
 }
 
 fn test_n_keys(factory: KamtFactory) {
+    const KEY_LEN: usize = 32;
     let store = MemoryBlockstore::default();
     // Test increasing numbers of sequential keys.
-    for i in 0u8..=255 {
-        let mut kamt: HKamt<_, _, _> = factory.new(&store);
-        for j in 0..i {
-            let mut k = [0; 32];
-            k[31] = j;
-            kamt.set(k, format!("{i}")).unwrap();
+    fn key(j: u64) -> [u8; KEY_LEN] {
+        let mut k = [0; KEY_LEN];
+        let encoded = j.to_be_bytes();
+        k[(KEY_LEN - encoded.len())..].copy_from_slice(&encoded[..]);
+        k
+    }
+
+    for do_flush in [true, false] {
+        for i in 0..=300 {
+            let mut kamt: HKamt<_, _, _> = factory.new(&store);
+            let k_too_big = key(i + 1);
+            for j in 0..i {
+                // Maybe try flushing/reloading (clearing the cache and/or dirty bits).
+                if do_flush {
+                    if j == i / 3 {
+                        // Flush but don't reload.
+                        kamt.flush().unwrap();
+                    } else if j == (2 * i) / 3 {
+                        // Flush and reload.
+                        let new_root = kamt.flush().unwrap();
+                        kamt.set_root(&new_root).unwrap();
```

Comment:
```suggestion
                        let new_root = kamt.flush().unwrap();
                        kamt.set(0, "yolo".to_string()).unwrap();
                        kamt.set_root(&new_root).unwrap();
```

don't need to accept these if you don't see the need, just seems like an opportunity to insert a lost mutation

Reply:
Eh, why not.